e2e/e2etests/test_add_observer.go

```diff
@@ -0,0 +1,200 @@
+package e2etests
+
+import (
+	"bytes"
+	"encoding/json"
+	"os/exec"
+	"strings"
+
+	sdkmath "cosmossdk.io/math"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/authz"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	"github.com/stretchr/testify/require"
+
+	"github.com/zeta-chain/node/e2e/runner"
+	"github.com/zeta-chain/node/e2e/utils"
+	"github.com/zeta-chain/node/pkg/parsers"
+	crosschaintypes "github.com/zeta-chain/node/x/crosschain/types"
+	emissionstypes "github.com/zeta-chain/node/x/emissions/types"
+	observertypes "github.com/zeta-chain/node/x/observer/types"
+)
+
+const (
+	StakeAmount = "1000000000000000000000azeta"
+)
+
+// AddNewObserver is a test function that adds a new observer to the network.
+func AddNewObserver(r *runner.E2ERunner) {
```

Comment:
I don't think this is the right place to add this.

It doesn't implement the E2E test interface and might be confusing to devs.

I would prefer keeping `e2etests` clean, with function you add in the e2e lists. This should rather belong to the runner package. It's run as a routine part of the igration test but not a test itself